gateway/enforcer/org.wso2.apk.enforcer.commons/src/main/java/org/wso2/apk/enforcer/commons/jwtgenerator/APIMgtGatewayJWTGeneratorImpl.java

```diff
@@ -110,6 +110,12 @@ public Map<String, Object> populateCustomClaims(JWTInfoDto jwtInfoDto) {
                 }
             }
         }
+        Map<String, String> customClaimsAPI = jwtInfoDto.getClaims();
+        if(customClaimsAPI != null) {
+            for (Map.Entry<String, String> customClaimEntry : customClaimsAPI.entrySet()) {
+                claims.putIfAbsent(customClaimEntry.getKey(), customClaimEntry.getValue());
```

Comment:
Shouldn't we override the existing claims with the custom claims? @tharindu1st

Reply:
yes we need to override the claims.

Reply:
Will fix this

Reply:
Fixed via [58c791f](https://github.com/wso2/apk/pull/1362/commits/58c791f238a73e1eb3523e8ddc5b99bdb202c017)